src/uptane/uptanerepository.cc

```diff
@@ -90,9 +92,13 @@ std::pair<uint32_t, std::vector<Uptane::Target> > Repository::getTargets() {
         throw MissMatchTarget("director and image");
```

Comment:
I'm struggling to understand why it is necessary to split the targets between the primary and secondary(s) here?

The split is only based on ecu_identifier, so would it make more sense to split them into two groups in SotaUptaneClient::runForever instead?